src/main/java/teammates/ui/controller/InstructorSearchPageData.java

```diff
@@ -58,4 +139,272 @@ public String getStudentRecordsLink(String courseId, StudentAttributes student){
+        
+        for (String fsName : feedbackResponseCommentSearchResultBundle.questions.keySet()) {
+            String courseId = feedbackResponseCommentSearchResultBundle.sessions.get(fsName).courseId;
+            
+            rows.add(new FeedbackSessionRow(fsName, courseId, createQuestionTables(fsName)));
+        }
+        return rows;
+    }
+    
+    private List<QuestionTable> createQuestionTables(String fsName) {
+        List<QuestionTable> questionTables = new ArrayList<QuestionTable>();
+        List<FeedbackQuestionAttributes> questionList = feedbackResponseCommentSearchResultBundle.questions.get(fsName);
+        
+        for (FeedbackQuestionAttributes question : questionList) {
+            int questionNumber = question.questionNumber;
+            String questionText = question.getQuestionDetails().questionText;
+            String additionalInfo = question.getQuestionDetails().getQuestionAdditionalInfoHtml(questionNumber, "");
+            
+            questionTables.add(new QuestionTable(questionNumber, questionText, additionalInfo, createResponseRows(question)));
+        }
+        return questionTables;
+    }
+    
+    private List<ResponseRow> createResponseRows(FeedbackQuestionAttributes question) {
+        List<ResponseRow> rows = new ArrayList<ResponseRow>();
+        List<FeedbackResponseAttributes> responseList = feedbackResponseCommentSearchResultBundle.responses.get(question.getId());
+        
+        for (FeedbackResponseAttributes responseEntry : responseList) {
+            String giverName = feedbackResponseCommentSearchResultBundle.responseGiverTable.get(responseEntry.getId());
+            String recipientName = feedbackResponseCommentSearchResultBundle.responseRecipientTable.get(responseEntry.getId());
+            String response = responseEntry.getResponseDetails().getAnswerHtml(question.getQuestionDetails());
+            
+            rows.add(new ResponseRow(giverName, recipientName, response, createFeedbackResponseCommentRows(responseEntry)));
+        }
+        return rows;
+    }
+    
+    private List<CommentRow> createCommentRows(String giverEmailPlusCourseId) {
+        List<CommentRow> rows = new ArrayList<CommentRow>();
+        
+        for (CommentAttributes comment : commentSearchResultBundle.giverCommentTable.get(giverEmailPlusCourseId)) {
+            String giverDetails = commentSearchResultBundle.giverTable.get(giverEmailPlusCourseId);
+            String recipientDetails = commentSearchResultBundle.recipientTable.get(comment.getCommentId().toString());
+            String creationTime = TimeHelper.formatTime(comment.createdAt);          
+            String link = getInstructorCommentsLink() + "&" + Const.ParamsNames.COURSE_ID 
+                                            + "=" + comment.courseId + "#" + comment.getCommentId();           
+            ElementTag editButton = createEditButton(link, Const.Tooltips.COMMENT_EDIT_IN_COMMENTS_PAGE);
```

Comment:
May be good to see which parts of these can be extracted out of the `for` loop.
For example, `giverDetails` will not change at all throughout the iterations, so there's no point searching for it in the list over and over again for the same value.